src/main/java/karuna/karuna_backend/receiver/domain/ReceiverService.java

```diff
@@ -0,0 +1,54 @@
+package karuna.karuna_backend.receiver.domain;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import karuna.karuna_backend.receiver.dto.ReceiverDTO;
+import karuna.karuna_backend.receiver.dto.ReceiversDTO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.concurrent.TimeUnit;
+
+@Service
+@RequiredArgsConstructor
+public class ReceiverService {
+
+    private final ReceiverRepository receiverRepository;
+
+    private final LoadingCache<String, ReceiversDTO> cache = Caffeine.newBuilder().expireAfterWrite(1, TimeUnit.HOURS).build(this::getAllReceiversFromDatabase);
```

Comment:
its to discuss, to push all to memory, its not be often used, and memor can be overflow

Reply:
@BarSwi what u think?

Reply:
Or acually since you are using cache for deleting users as well we might create cache for shorte duration? Like 5-10 minutes.